R/defStormsDataset.R

```diff
@@ -241,23 +248,350 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+                 dim = c(row, len)
+    )
+  } else if (unitConversion["msw"] == "knt2ms") {
+    msw <- array(knt2ms(ncdf4::ncvar_get(dataBase, fields["msw"])[, ind]),
+                 dim = c(row, len)
+    )
+  } else if (unitConversion["msw"] == "kmh2ms") {
+    msw <- array(kmh2ms(ncdf4::ncvar_get(dataBase, fields["msw"])[, ind]),
+                 dim = c(row, len)
+    )
+  } else {
+    msw <- array(ncdf4::ncvar_get(dataBase, fields["msw"])[, ind],
+                 dim = c(row, len)
+    )
+  }
+  
+  # Collect data
+  data <- list(
+    names = ncdf4::ncvar_get(dataBase, fields["names"])[ind],
+    seasons = season[ind],
+    isotimes = array(ncdf4::ncvar_get(dataBase, fields["isoTime"])[, ind],
+                     dim = c(row, len)
+    ),
+    longitude = array(ncdf4::ncvar_get(dataBase, fields["lon"])[, ind],
+                      dim = c(row, len)
+    ),
+    latitude = array(ncdf4::ncvar_get(dataBase, fields["lat"])[, ind],
+                     dim = c(row, len)
+    ),
+    msw = msw
+  )
+  
+  # Sort by Date
+  o <- order(data$isotimes[1, ])
+  
+  data$names <- data$names[o]
+  data$seasons <- data$seasons[o]
+  data$isotimes <- data$isotimes[, o]
+  data$longitude <- data$longitude[, o]
+  data$latitude <- data$latitude[, o]
+  data$msw <- data$msw[, o]
+  
+  if ("sshs" %in% names(fields)) {
+    data$sshs <- array(ncdf4::ncvar_get(dataBase, fields["sshs"])[, ind], dim = c(row, len))
+    data$sshs <- data$sshs[, o]
+  }
+  
+  if ("rmw" %in% names(fields)) {
+    if (unitConversion["rmw"] == "nm2km") {
+      data$rmw <- array(nm2km(ncdf4::ncvar_get(dataBase, fields["rmw"])[, ind]), dim = c(row, len))
+    } else {
+      data$rmw <- array(ncdf4::ncvar_get(dataBase, fields["rmw"])[, ind], dim = c(row, len))
+    }
+    data$rmw <- data$rmw[, o]
+  }
+  
+  
+  if ("pressure" %in% names(fields)) {
+    if (unitConversion["pressure"] == "mb2pa") {
+      data$pressure <- array(mb2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "b2pa") {
+      data$pressure <- array(b2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "psi2pa") {
+      data$pressure <- array(psi2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "atm2pa") {
+      data$pressure <- array(atm2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else {
+      data$pressure <- array(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind], dim = c(row, len))
+    }
+    data$pressure <- data$pressure[, o]
+  }
+  
+  if ("poci" %in% names(fields)) {
+    if (unitConversion["poci"] == "mb2pa") {
+      data$poci <- array(mb2pa(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["poci"] == "b2pa") {
```

Comment:
Could you add tests for all these 3 cases (was this a bug not previously handled ?)

Reply:
This was not a bug, it was not there before, and it should have